kernel/src/hil/nfc.rs

```diff
@@ -0,0 +1,105 @@
+use crate::returncode::ReturnCode;
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+/// An enum to keep track of the NFC field status
+pub enum NfcFieldState {
+    /// Initial value that indicates no NFCT field events.
+    None,
+    /// The NFCT FIELDLOST event has been set.
+    On,
+    /// The NFCT FIELDDETECTED event has been set.
+    Off,
```

Comment:
I'm a bit confused by the words `On` and `Off` here, as compared to the documentation. If I understand correctly, this driver is exclusively for a NFC edge device (i.e. it cannot turn on and generate a field, but it can respond to one when it finds it).

Would this maybe better expressed as a `NfcFieldPresent` enum?